### PR TITLE
Hvdc deletion including shunt compensators

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java [hunk 1]
```diff
@@ -133,6 +133,16 @@ public String getEquipmentMapData(UUID networkUuid, String variantId, String equ
...
 
+    public String getHvdcLineWithShuntCompensators(UUID networkUuid, String variantId, String hvdcId) {
+        UriComponentsBuilder builder = UriComponentsBuilder.fromPath(
+                DELIMITER + NETWORK_MAP_API_VERSION + "/networks/{networkUuid}/hvdc-lines/{hvdcId}/shunt-compensators");
+        if (!StringUtils.isBlank(variantId)) {
+            builder = builder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
+        }
+        String path = builder.buildAndExpand(networkUuid, hvdcId).toUriString();
+        return restTemplate.getForObject(networkMapServerBaseUri + path, String.class);
+    }
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:144]: Add `handleHttpError`

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java [hunk 2]
```diff
@@ -133,6 +133,22 @@ public String getEquipmentMapData(UUID networkUuid, String variantId, String equ
...
+        UriComponentsBuilder builder = UriComponentsBuilder.fromPath(
+                DELIMITER + NETWORK_MAP_API_VERSION + "/networks/{networkUuid}/hvdc-lines/{hvdcId}/shunt-compensators");
+        if (!StringUtils.isBlank(variantId)) {
+            builder = builder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
+        }
+        String path = builder.buildAndExpand(networkUuid, hvdcId).toUriString();
+        String equipmentMapData;
+        try {
+            equipmentMapData = restTemplate.getForObject(networkMapServerBaseUri + path, String.class);
+        } catch (HttpStatusCodeException e) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:146]: if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
                throw new StudyException(EQUIPMENT_NOT_FOUND);
            } else {
                throw handleHttpError(e, GET_NETWORK_ELEMENT_FAILED);
            }

Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:146]: we may want to get infos on an unexisting Hvdc line (can happen in a modification). Then the endpoint cannot return NOT_FOUND.

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -323,6 +323,18 @@ public ResponseEntity<List<IdentifiableInfos>> getVoltageLevelBusbarSections(
...
 
+    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/hvdc-lines/{hvdcId}/shunt-compensators")
+    @Operation(summary = "For a given hvdc line, get its related Shunt compensators in case of LCC converter station")
+    @ApiResponse(responseCode = "200", description = "Hvdc line type and its shunt compensators on each side")
+    public ResponseEntity<String> getHvdcLineWithShuntCompensators(
+            @PathVariable("studyUuid") UUID studyUuid,
+            @PathVariable("nodeUuid") UUID nodeUuid,
+            @PathVariable("hvdcId") String hvdcId,
+            @Parameter(description = "Should get in upstream built node ?") @RequestParam(value = "inUpstreamBuiltParentNode", required = false, defaultValue = "true") boolean inUpstreamBuiltParentNode) {
+        String hvdcInfos = studyService.getHvdcLineWithShuntCompensators(studyUuid, nodeUuid, inUpstreamBuiltParentNode, hvdcId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:334]: Rename by `getHvdcLineShuntCompensators` ?

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java [hunk 4]
```diff
@@ -133,6 +133,22 @@ public String getEquipmentMapData(UUID networkUuid, String variantId, String equ
         return equipmentMapData;
     }
 
+    public String getHvdcLineWithShuntCompensators(UUID networkUuid, String variantId, String hvdcId) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:136]: Rename ,

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:146]: ok

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 5]
```diff
@@ -673,6 +673,13 @@ public String getVoltageLevelEquipments(UUID studyUuid, UUID nodeUuid, List<Stri
                 substationsIds, equipmentPath);
     }
 
+    public String getHvdcLineWithShuntCompensators(UUID studyUuid, UUID nodeUuid, boolean inUpstreamBuiltParentNode, String hvdcId) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:676]: Rename ?

File: src/test/java/org/gridsuite/study/server/NetworkMapTest.java [hunk 6]
```diff
@@ -158,18 +157,21 @@ public MockResponse dispatch(RecordedRequest request) {
...
                                 .addHeader("Content-Type", "application/json; charset=utf-8");
                     case "/v1/networks/" + NETWORK_UUID_STRING + "/voltage-levels-equipments":
                         return new MockResponse().setResponseCode(200).setBody(VOLTAGE_LEVELS_EQUIPMENTS_JSON)
                                 .addHeader("Content-Type", "application/json; charset=utf-8");
-
                     case "/v1/networks/" + NETWORK_UUID_STRING + "/voltage-level-equipments/" + VL_ID_1:
                         return new MockResponse().setResponseCode(200).setBody(VOLTAGE_LEVEL_EQUIPMENTS_JSON)
                                 .addHeader("Content-Type", "application/json; charset=utf-8");
-
+                    case "/v1/networks/" + NETWORK_UUID_STRING + "/hvdc-lines/" + HVDC_LINE_ID_1 + "/shunt-compensators":
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/NetworkMapTest.java:169]: Why you do not use `WireMock` ?

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**